### ISSUE TITLE
composer.required.json/composer.suggested.json are git ignored

### ISSUE THREAD

**Issue by joachim-n** [acquia/blt#3196]
BLT adds a blt/.gitignore file which contains:

```
composer.required.json
composer.suggested.json
```

This means that if developer A installs BLT and commits it to the project repository, and developer B pulls that code, they don't get those two files. That then causes composer to fail validation because the files are referenced in the composer.json merge-plugin section.

**Comment by malikkotob** [comment]
@joachim-n I would assume that when developer B clones the repo and runs `composer install`, that BLT would place those same files in developer B's local project repo, is that not the case?

**Comment by ba66e77** [comment]
Normally the vendor directory is not committed to your git repo, only the composer.json and composer.lock. Developers would run `composer install` to build that vendor directory

**Comment by joachim-n** [comment]
I've been on multiple projects where that's not been the case!

**Comment by ba66e77** [comment]
Yeah, I know some projects _do_ commit the vendor directory, but BLT's assumption is that vendor is ignored and we list it in the .gitignore we ship.

**Comment by joachim-n** [comment]
AFAIK those two JSON files, composer.required.json, are copied verbatim from /vendor/acquia/blt to /blt, and should not be altered be project developers.

Would it not be possible to have the root composer.json's merge-plugin section refer to /vendor/acquia/blt/composer.required.json, and not do the copy operation? You'd then support both methodologies for dealing with Composer packages in a project repository.

**Comment by wizonesolutions** [comment]
I changed mine back to the vendor copies because I couldn't find a sane way to initialize them locally (or, in my case, on a build server that simply pulled down the repo and `composer install`ed. It was getting hung up because those files did not exist on it, and it could not `composer install` either because I had upgraded BLT in the meantime. Maybe BLT needs some scripts to handle these Composer-broken cases.
